smartrent/distraught
# add raven to exports and set tags for each module's raven.config call

## Description

### Issue:

It is difficult to trace sentry events back to either the web or cron servers for supportrent
- Testing the sentry config changes in support PR https://github.com/smartrent/support/pull/604 are difficult to validate frontend vs. backend sentry events
- Current work around is to filter issues by `sdk.name:raven-node` to know it originated from distraught's sentry instance
- Sometimes hard to tell if the error came from web or cron servers.

### Solution:

We add `raven` to distraught's exports
- Gives downstream apps access to distraught's configured raven installation
- Allows adding custom tags to using [`raven.config`](https://raven-js.readthedocs.io/en/stable/usage.html)
- We can add / test any custom tags we want to add in downstream apps & not have to update distraught if we want to change the tags later

### Example:

We can use this code to add custom tags in support:
```
D.raven.config({ tags: { jsPackage: "BACKEND", jsModule: "web" } });
```

---

## Steps to test
1. run `yarn build`
2. replace support api's node_module dist folder with the build in ./dist folder
  a. go to support project
  b. run `cd api && yarn install`
  c. replace this folder `api/node_modules/@smartrent/distraught/dist`
3. add this code to end of the `api/router.ts` file
```
console.log("BEFORE", D.raven["_globalContext"]);
D.raven.config({ tags: { jsPackage: "BACKEND", jsModule: "web" } });
console.log("AFTER", D.raven["_globalContext"]);
throw new Error(
  "this is a test to see if raven tags show up in sentry for this error event"
);
```
4. start support api server for web
5. the server should startup and you should the error we added in the console
6. should see error in [sentry ](https://sentry.io/organizations/smartrent/issues/?project=1770380)
- Example:
  -  https://sentry.io/organizations/smartrent/issues/2653446761/?referrer=slack


## Screenshots
![953CD771-C57F-43E0-BB30-384F543AB2F1](https://user-images.githubusercontent.com/34978297/133657057-28d3d77a-859f-419f-858f-05b0952fed68.png)
![9459EA5A-6E71-455D-993E-A86C0599AC44](https://user-images.githubusercontent.com/34978297/133657125-5a030d3c-6e17-4929-b802-66ef1f4273a2.png)
